### ISSUE TITLE
HistoryReader Error

### ISSUE THREAD

**Issue by rabyunghwa** [ben-xo/sslscrobbler#18]
When I tried to load history_reader.php into cmd, it showed me some fatal errors (as shown in the screenshot). So what could possibly go wrong? Thanks in advance.
![capture](https://cloud.githubusercontent.com/assets/7411385/10158921/8cb9e488-66c7-11e5-8677-b465f7a6d7e5.PNG)

**Comment by ben-xo** [comment]
Hi there,

You need to cd to Users\ByubgHwa\Desktop first. The clue is right there on
the screen. :)

**Comment by rabyunghwa** [comment]
Hey dude, I did as you said. This time it didn't show any errors. The php file just pops out. Is this the expected behavior? Then I moved onto the next step, which is to start scrobbling by typing in the cmd "php historyreader.php -L lastfmusername", then the same errors shows up again. Is there any more specific guides that I can follow? I'm quite a newbie and sometimes can feel really confused following the descriptions. Also I've found that this library hasn't been maintained for quite a long period. Is there any other solution already? Thanks in advance.
![capture](https://cloud.githubusercontent.com/assets/7411385/10261681/4e3add5e-69d6-11e5-9a8f-5a519d42fe16.PNG)

**Comment by ben-xo** [comment]
I'm afraid it might be time for to learn a bit more about the command
prompt - I'm not sure how to explain it better without actually showing you
what to do (rather than just describing it).

Running "historyreader" with no options doesn't do anything, you need to
add the options all in the same command

**Comment by rabyunghwa** [comment]
Hey dude. I think I made some progress on the use of cmd prompt. However, there seems to be some other problem going on. So please help me dude. FYI, my OS version is Win10.
![capture](https://cloud.githubusercontent.com/assets/7411385/10423367/738ed456-70f8-11e5-82a0-d92bc9301f14.PNG)

**Comment by rabyunghwa** [comment]
Hey dude I've figured it out! 
https://serato.com/scratchlive
Like what's said there, scratchlive has been updated to Serato so will this still work?
Thanks.

**Comment by ben-xo** [comment]
Hi, the problem is that your PHP doesn't have the multibyte (mb) plugin
enabled.

**Comment by rabyunghwa** [comment]
Dude thank you very much. However, there's new problems arising. It says String could not be parsed as XML. So is there anything that I have not installed? I've done everything specified in your instructions. Also I could not tweet. Please help me with this. Thanks.
![capture](https://cloud.githubusercontent.com/assets/7411385/10431415/7a3b713a-7137-11e5-8c1b-456dda949eb6.PNG)

**Comment by ben-xo** [comment]
Hmm! That looks like it going be a genuine bug, or a problem with Last.fm…
Good job I work for Last.fm. I will look into that.

**Comment by rabyunghwa** [comment]
Dude I'm also having some problem inserting the playlist into MySQL database. More specifically, about the MySQL setup in the php.ini file. Can you elaborate more on this?

**Comment by ben-xo** [comment]
The issue "XML: Extra content at the end of the document" is now resolved (it was a bug with Last.fm! Fix made at Last.fm end) which means you can now connect to Last.fm.

**Comment by ben-xo** [comment]
Regarding MySQL setup: unless you have a specific need for this, don't bother. That was added for a specific user who asked for it in order to integrate with a radio station application. If you don't know how to set it up, you probably don't need it.

**Comment by rabyunghwa** [comment]
Hey dude thank you very much for the fix! It's working now! This is great! Thanks again!
But dude I'm afraid I'll need your help with the MySQL setup. I'm currently working on an Android client app where the currently playing song's info like its name is displayed. What I have in mind is insert the song name into MySQL database when the underlying observer is triggered. And then push a message or a notification from the server to my Android client app. Do you think this is the right approach? Do you know if there is any RSS feed from Serato website that I can fetch from which I can get the now-playing song's info? It would be much easier if there were any external API that I can use to get it. Thank you very much.

**Comment by rabyunghwa** [comment]
Dude I tried the MySQL implementation but to no avail. Could you please help me with it? Can't seem to get the configuration right. Thank you very much.
![capture](https://cloud.githubusercontent.com/assets/7411385/10562447/2181f8e6-758f-11e5-945e-15d4100290ad.PNG)

**Comment by ben-xo** [comment]
You need to enable the MySQL module for PHP in the .ini file the same way
you enabled mb multibyte module.

**Comment by rabyunghwa** [comment]
Yes dude I did enable the module but not sure if it's correct. Where do I set the specific database and the table to insert the data into? 
![save1](https://cloud.githubusercontent.com/assets/7411385/10562483/053c4018-7591-11e5-8fea-a7d0a3bd1255.PNG)
![save2](https://cloud.githubusercontent.com/assets/7411385/10562486/09b78166-7591-11e5-84d8-6c256c767a09.PNG)
![save3](https://cloud.githubusercontent.com/assets/7411385/10562488/0dbf816e-7591-11e5-9d3e-496f967f5d47.PNG)

**Comment by ben-xo** [comment]
The error your seeing there - I think you need the other MySQL driver.
There are two.

You set the details in the config.php - there is an example commented out
in there, I think! If not I'll give you an example later.

**Comment by rabyunghwa** [comment]
Hey dude please give me an example. I changed the file's name from "config.php-default" to "config.php" and then modified it. But still it's not working.
![capture](https://cloud.githubusercontent.com/assets/7411385/10563530/434c6e80-75c0-11e5-86c4-e2b79e93e8c2.PNG)
![capture2](https://cloud.githubusercontent.com/assets/7411385/10563531/4d6f3e10-75c0-11e5-8778-3bb68d00eb05.PNG)

**Comment by rabyunghwa** [comment]
Dude, sorry that I have so many questions to ask. It's just that I'm quite new to PHP and stuff. Really appreciate your help and patience

**Comment by ben-xo** [comment]
Hey there,

The fact that it says "db plugin could not find driver" I think means you
still need to uncomment the correct MySQL plugin for php. You still need to
solve that problem.

However you also still have the default database name "hello" in your
config...

**Comment by rabyunghwa** [comment]
Okay dude thank you very much! I think I'm getting closer and closer to what I want to achieve! However, nothing is inserted into my database despite the fact that this time no error is shown. Also I'm a little confused about what the 'user' and 'key' refer to in the statement '...WHERE user=:key'. Could you please just give me a working example? I'd really appreciate your help.
![capture](https://cloud.githubusercontent.com/assets/7411385/10564785/5fd93582-75f3-11e5-938a-3e029ad74afa.PNG)

**Comment by rabyunghwa** [comment]
Also dude I just found out that Scratch Live had been replaced by SeratoDJ. And I tried it but it seemed that sslscrobbler would not work with SeratoDJ. Does it have to do with the folder where sslscrobbler finds the history session file being different? If this is the problem, then will you bring any update in the future to support SeratoDJ? Thanks.

**Comment by ben-xo** [comment]
Hi there, as far as I was aware it works just fine with seratodj.

I'm sorry I don't think I can offer you more help with the MySQL plugin.
When I said it was only for those who know what they need, I think you
should probably refer this o your database administrator or someone else
who knows SQL.

**Comment by dfraser** [comment]
It does.  I use it weekly!

--
Dan Fraser

**Comment by rabyunghwa** [comment]
Hey dude okay. Thank you so much for your generous help so far. And Dan, if you could help me with it, I would really appreciate it!

**Comment by rabyunghwa** [comment]
Hey dudes! It works with SeratoDJ!

**Comment by ben-xo** [comment]
@rabyunghwa - ten years late replying to this but i'm glad it worked for you (and indeed I use it with Serato DJ every week myself!!)
